iroh-store/src/rpc.rs

```diff
@@ -0,0 +1,101 @@
+use std::io::Cursor;
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+// use bytes::Bytes;
+use cid::Cid;
+use eyre::Result;
+use futures::lock::Mutex;
+use iroh_rpc_types::store::store_server;
+use iroh_rpc_types::store::{
+    GetLinksRequest, GetLinksResponse, GetRequest, GetResponse, PutRequest,
+};
+use tonic::{transport::Server as TonicServer, Request, Response, Status};
+use tracing::info;
+
+use crate::store::InnerStore;
+
+struct Rpc {
+    store: Arc<Mutex<InnerStore>>,
+}
+
+#[tonic::async_trait]
+impl store_server::Store for Rpc {
+    async fn put(&self, request: Request<PutRequest>) -> Result<Response<()>, tonic::Status> {
+        let req = request.into_inner();
+        let cid = cid_from_bytes(req.cid)?;
+        let links = links_from_bytes(req.links)?;
+        let res = self
+            .store
+            .lock()
+            .await
+            .put(cid, req.blob, links)
+            .await
+            .map_err(|e| Status::internal(format!("{:?}", e)))?;
+
+        info!("store rpc call: put cid {}", cid);
+        Ok(Response::new(res))
+    }
+
+    async fn get(
+        &self,
+        request: Request<GetRequest>,
+    ) -> Result<Response<GetResponse>, tonic::Status> {
+        let req = request.into_inner();
+        let cid = cid_from_bytes(req.cid)?;
+        if let Some(res) = self
+            .store
+            .lock()
+            .await
+            .get(&cid)
+            .await
+            .map_err(|e| Status::internal(format!("{:?}", e)))?
+        {
+            todo!("return DBPinnableSlice as Bytes")
+            // Ok(Response::new(GetResponse {
+            //     data: Some(Bytes::from(res)),
+            // }))
```

Comment:
How do I go from `DBPinnableSlice` to `bytes::Bytes`?

I find myself in this problem a bunch, where I know there should be a way to convert things but I don't know how to connect the dots.

Would love to talk about what I might be missing in my thinking so that I don't keep repeating the same problem.